vagrant/vagrant-scripts/vagrant-up.sh

```diff
@@ -1,9 +1,41 @@
 #!/bin/bash
 
-set -euo pipefail
+set -eo pipefail
 
 export K8S_NODE_OS=${K8S_NODE_OS:-ubuntu}
 export K8S_NODES=${K8S_NODES:-1}
 export VAGRANT_DEFAULT_PROVIDER=${VAGRANT_DEFAULT_PROVIDER:-virtualbox}
 
-vagrant up 
+# Default values for environment deployment
+DEV_ENV="false"
+TEST_ENV="true"
+
+# Override defaults from arguments
+while [ "$1" != "" ]; do
+    case $1 in
+        -d | --dev-env )
+            DEV_ENV="true"
+            echo "Using development environment"
+            ;;
+        -t | --test-env )
+            TEST_ENV="true"
+            echo "Using testing environment"
+            ;;
+        * )
+            echo "Invalid parameter: "$1
+            exit 1
+    esac
+    shift
+done
+
+if [ "${DEV_ENV}" == "true" ]
+then
+    cp ../Vagrantfile-dev ../Vagrantfile
+    vagrant up
```

Comment:
you can set `VAGRANT_VAGRANTFILE` environment variable to point the vagrant file, instead of cp and rm later
`VAGRANT_VAGRANTFILE =foobar vagrant up`